File: julia/2017/5/solution2.jl
f = open("../../../data/2017/5/data.txt")
vals = map(x -> parse(Int,x), readlines(f))
close(f)

count = 0
ind = 1
while true
    old_val = vals[ind]
    if old_val >= 3
        vals[ind] -= 1
    else
        vals[ind] += 1
    end
    count += 1
    ind += old_val
    if ind > length(lines)
        println("Solution is $count")
        break
    end
end

File: scala/2017/5/solution2.scala
import scala.io.Source
import scala.util.control.Breaks._

var vals = Source.fromFile("../../../data/2017/5/data.txt").getLines.toList.map(_.toInt)

var count: Int = 0
var ind: Int = 0
var old_val: Int = 0

while (true) {
  old_val = vals(ind)
  if (old_val >= 3) {
    vals = vals.updated(ind, old_val - 1)
  } else {
    vals = vals.updated(ind, old_val + 1)
  }
  count += 1
  ind += old_val
  if (ind >= vals.length) {
    println(f"Solution is $count")
    break
  }
}
